src/base/emu-i386/simx86/interp.c

```diff
@@ -251,22 +251,14 @@ static unsigned int _JumpGen(unsigned int P2, int mode, int cond,
 		    dbug_printf("!Forever loop!\n");
 		    leavedos_main(0xebfe);
 		}
-#ifdef HOST_ARCH_X86
-		/* note: nojumps is disabled by default -- while it seems
-		   an easy optimization, it forces re-JITting of the code
-		   at the jmp's target, instead of using the node linker
-		   to see if there is already code there */
-#ifdef NOJUMPS
 #if !defined(SINGLESTEP)
-		if (!(EFLAGS & TF) && !CONFIG_CPUSIM &&
+		if (CONFIG_CPUSIM && !(EFLAGS & TF) &&
```

Comment:
Maybe this breaks SIM?

Reply:
You're correct about the sim breakage but that puzzled me, so I looked a bit further. The real issue here is the `return` in the `JumpGen` macro, which causes an exit from `Interp86()`, instead of not executing CloseAndExec. With a change as follows that is avoided:
```
@@ -297,11 +294,11 @@ static unsigned int _JumpGen(unsigned int P2, int mode, int cond,
        unsigned int _P0, _P1; \
        int _rc; \
        _P1 = _JumpGen(P2, mode, cond, btype, &_P0); \
-       if (_P1 != (unsigned)-1) { \
-               return _P1; } \
-       if (!CONFIG_CPUSIM) \
-               NewIMeta(P2, mode, &_rc); \
-       _P1 = CloseAndExec(_P0, mode, __LINE__); NewNode=0; \
+       if (_P1 == (unsigned)-1) { \
+               if (!CONFIG_CPUSIM) \
+                       NewIMeta(P2, mode, &_rc); \
+               _P1 = CloseAndExec(_P0, mode, __LINE__); NewNode=0; \
+       } \
        _P1; \
 })

```
(Using git blame I can see that that is a regression caused by abb05e0).

Now as for the uses of `TheCPU.eip`: they should really all be deleted or replaced by local variables in interpreted opcodes, as it's always set when exiting `Interp86`.

Now my question is how much does the above change on its own help Spear of Destiny?